common/src/java/org/apache/hadoop/hive/common/FileUtils.java

```diff
@@ -483,12 +483,6 @@ public static boolean isActionPermittedForFileHierarchy(FileSystem fs, FileStatu
       String userName, FsAction action, boolean recurse) throws Exception {
     boolean isDir = fileStatus.isDir();
 
-    FsAction dirActionNeeded = action;
-    if (isDir) {
-      // for dirs user needs execute privileges as well
-      dirActionNeeded.and(FsAction.EXECUTE);
-    }
-
```

Comment:
So, I understand why this would be removed from a find-bugs perspective (this is a no-op), but this is actually an all around bug.  This should be:

```
// for dirs user needs execute privileges as well
FsAction dirActionNeeded = (isDir) ? action.and(FsAction.EXECUTE) : action;
```